tasks/carto.py

```diff
@@ -1399,15 +933,151 @@ def run(self):
                         dimension=dimension
                     ))
             session.commit()
-            self._complete = True
         except:
             session.rollback()
             session.execute('DROP TABLE IF EXISTS observatory.obs_meta_next')
             session.commit()
             raise
 
     def complete(self):
-        return getattr(self, '_complete', False)
+        tables = ['obs_meta', 'obs_meta_numer', 'obs_meta_denom',
+                  'obs_meta_geom', 'obs_meta_timespan']
+        # Only look into whether new denormalized tables would be different
```

Comment:
Ok. After some time struggling with this in my local install, I think that I understand what's happening. I wanted to test catalog generation, and I thought that `make clean-catalog` and then `make catalog` would recreate it. I think that this is not longer true with this change. This change make `complete` return `False`if there's no change between "denormalized tables and the old ones", so it's marked as not completed in the pipeline, thus making `GenerateRST` fail because of unfulfilled dependencies:

```
INFO: [pid 1] Worker Worker(salt=143001522, workers=1, host=a25737667a9b, username=root, pid=1) running   GenerateRST(format=html, section=None)
> /bigmetadata/tasks/carto.py(986)complete()
-> return False
(Pdb) c
ERROR: [pid 1] Worker Worker(salt=143001522, workers=1, host=a25737667a9b, username=root, pid=1) failed    GenerateRST(format=html, section=None)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/luigi/worker.py", line 161, in run
    raise RuntimeError('Unfulfilled %s at run time: %s' % (deps, ', '.join(missing)))
RuntimeError: Unfulfilled dependency at run time: OBSMetaToLocal_False_e6030a9882
INFO: Skipping error email. Set `error-email` in the `core` section of the luigi config file or override `owner_email`in the task to receive error emails.
INFO: Informed scheduler that task   GenerateRST_html_None_969fc8a5f6   has status   FAILED
INFO: Starting pruning of task graph
INFO: Done pruning task graph
INFO: Worker Worker(salt=143001522, workers=1, host=a25737667a9b, username=root, pid=1) was stopped. Shutting down Keep-Alive thread
INFO:
===== Luigi Execution Summary =====

Scheduled 4 tasks of which:
* 1 present dependencies were encountered:
    - 1 ConfirmTablesDescribedExist()
* 1 ran successfully:
    - 1 OBSMetaToLocal(force=False)
* 1 failed:
    - 1 GenerateRST(format=html, section=None)
* 1 were left pending, among these:
    * 1 had failed dependencies:
        - 1 Catalog(format=html, section=None)

This progress looks :( because there were failed tasks
```

I'll continue taking a look at this. Maybe `complete` behaviour should change?

cc @rafatower @ethervoid